### PR TITLE
Update core modules - flexibleserver and appservice to support managed Identity

### PR DESCRIPTION
**Pull request by charris-msft**
postgres flexibleserver:
  - add authConfig, backup and maintenanceWindow

appservice:
  - add httpLoggingEnabled
  - add 'azd-service-name': 'web' to tags

### CODE REVIEW

File: templates/common/infra/bicep/core/host/appservice.bicep [hunk 1]
```diff
@@ -33,19 +33,21 @@ param numberOfWorkers int = -1
...
 param use32BitWorkerProcess bool = false
 param ftpsState string = 'FtpsOnly'
+param httpLoggingEnabled bool = false
 param healthCheckPath string = ''
 
 resource appService 'Microsoft.Web/sites@2022-03-01' = {
   name: name
   location: location
-  tags: tags
+  tags: union(tags, { 'azd-service-name': 'web' })
```
Comment (ellismg) [inline templates/common/infra/bicep/core/host/appservice.bicep:42]: This doesn't seem right - these core modules should not contain `azd` specific tags.  This module might be used by multiple services and so the `azd-service-name` tag should be passed into this module (as a member of it's `tags` property)

**Review by ellismg — CHANGES_REQUESTED** [review]
A little concerned about the app service changes.

File: templates/common/infra/bicep/core/host/appservice.bicep (hunk 1 above)
Comment (jongio) [reply templates/common/infra/bicep/core/host/appservice.bicep:42]: matt is correct

File: templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep [hunk 2]
```diff
@@ -29,6 +30,20 @@ resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-12-01' =
...
+    backup: {
+      backupRetentionDays: 7
+      geoRedundantBackup: 'Disabled'
+    }
+    maintenanceWindow: {
+      customWindow: 'Disabled'
+      dayOfWeek: 0
+      startHour: 0
+      startMinute: 0
+    }
```
Comment (jongio) [inline templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep:46]: These should be params with defaults so user can mod if they want

Comment (charris-msft) [reply templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep]: I've updated the PR with params.

Where the setting name wasn't as clear like `customWindow` I made the parameter `maintenanceWindowCustomWindow`, but I'm wondering about doing something more like 'maintenanceWindow_customWindow` to indicate the relationship more clearly.

Thoughts?

Comment (jongio) [reply templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep]: Could you look into: https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters#objects-as-parameters

**Pull request closed without merging**